sklearn/mixture/gaussian_mixture.py

```diff
@@ -0,0 +1,673 @@
+
+        param_shape = {'weights': (self.n_components,),
+                       'means': (self.n_components, self.n_features),
+                       'covariances': cov_shape[self.covariance_type]}
+
+        return param_shape
+
+    def _check_parameters(self):
+        """Check the Gaussian mixture parameters are well defined."""
+        if self.covariance_type not in ['spherical', 'tied', 'diag', 'full']:
+            raise ValueError("Invalid value for 'covariance_type': %s "
+                             "'covariance_type' should be in "
+                             "['spherical', 'tied', 'diag', 'full']"
+                             % self.covariance_type)
+
+        param_shape = self._define_parameter_shape()
+
+        if self.weights_init is not None:
+            self.weights_init = _check_weights(
+                self.weights_init, param_shape['weights'])
+
+        if self.means_init is not None:
+            self.means_init = _check_means(
+                self.means_init, param_shape['means'])
+
+        if self.covariances_init is not None:
+            self.covariances_init = _check_covariances(
+                self.covariances_init, param_shape['covariances'],
+                self.covariance_type)
+
+    def _initialize(self, X, resp):
+        """Initialization of the Gaussian mixtures parameters.
+
+        Parameters
+        ----------
+        X : array-like, shape = (n_samples, n_features)
+
+        resp : array-like, shape = (n_samples, n_components)
+        """
+        weights, means, covariances = _estimate_gaussian_parameters(
+            X, resp, self.reg_covar, self.covariance_type)
+        self.weights_ = (weights if self.weights_init is None
+                         else self.weights_init)
+        self.means_ = means if self.means_init is None else self.means_init
+        self.covariances_ = (covariances if self.covariances_init is None
+                             else self.covariances_init)
+
+    def _e_step(self, X):
+        log_prob_norm, resp = self._estimate_log_prob_resp(X)
+        self._log_likelihood = np.sum(log_prob_norm)
+        return self._log_likelihood, resp
+
+    def _m_step(self, X, resp):
+        self.weights_, self.means_, self.covariances_ = (
+            _estimate_gaussian_parameters(X, resp, self.reg_covar,
+                                          self.covariance_type))
+
+    def _estimate_log_prob(self, X):
+        estimate_log_prob_functions = {
+            "full": _estimate_log_gaussian_prob_full,
+            "tied": _estimate_log_gaussian_prob_tied,
+            "diag": _estimate_log_gaussian_prob_diag,
+            "spherical": _estimate_log_gaussian_prob_spherical
+        }
+        return estimate_log_prob_functions[self.covariance_type](
+            X, self.means_, self.covariances_)
+
+    def _estimate_log_weights(self):
+        return np.log(self.weights_)
+
+    def _check_is_fitted(self):
+        check_is_fitted(self, 'weights_')
```

Comment:
you can give a list `['weights_', 'means_', 'covariances']` to have only one call to `check_is_fitted`